### PR TITLE
orocos-kdl: init at 1.4.0

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Adds the [Orocos Kinematic and Dynamics library](https://www.orocos.org/kdl.html), along with its Python bindings (PyKDL). This library is an important dependency for newer versions of the Robot Operating System (ROS).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by IvarWithoutBones — APPROVED** [review]
Result of `nixpkgs-review pr 97998` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>orocos-kdl</li>
    <li>python27Packages.pykdl</li>
    <li>python37Packages.pykdl</li>
    <li>python38Packages.pykdl</li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -4976,6 +4976,8 @@ in {
 
   pyjwt = callPackage ../development/python-modules/pyjwt { };
 
+  pykdl = callPackage ../development/python-modules/pykdl { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:4979]:
```suggestion
  pykdl = toPythonModules (callPackage ../development/python-modules/pykdl { });
```

**Review by jonringer — COMMENTED** [review]
or else you will have issues importing the package

**Review by jonringer — APPROVED** [review]
LGTM

```
https://github.com/NixOS/nixpkgs/pull/97998
4 packages built:
orocos-kdl python27Packages.pykdl python37Packages.pykdl python38Packages.pykdl
```

**Pull request merged**